app/src/shared/components/Search/index.jsx

```diff
@@ -20,20 +21,30 @@ type State = {
 }
 
 class Search extends React.Component<Props, State> {
-    state = {
-        isOpen: this.props.value !== '',
-        text: this.props.value || '',
+    constructor(props: Props) {
+        super(props)
+        this.debouncedOnChange = debounce(props.onChange, 500)
+
+        this.state = {
+            isOpen: this.props.value !== '',
+            text: this.props.value || '',
+        }
     }
 
     onTextChange = (e: SyntheticInputEvent<EventTarget>) => {
         const { value } = e.target
-        this.props.onChange(value)
+
         this.setState({
             text: value,
         })
+
+        if (this.debouncedOnChange) {
```

Comment:
`this.debouncedOnChange` should be always available since it's assigned in the constructor?

Reply:
Oh, true. At one point Flow was complaining but not anymore 👍